### PR TITLE
reference: add more info for errors

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### What is changed, added or deleted? (Required)

Add more info for errors.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: reference/error-codes.md [hunk 1]
```diff
@@ -33,83 +33,75 @@ TiDB 兼容 MySQL 的错误码，在大多数情况下，返回和 MySQL 一样
...
 
 * Error Number: 8019
 
-    重新载入的插件版本与之前不同，无法重新载入 
+    重新载入的插件版本与之前不同，无法重新载入。重新载入插件，要求插件的版本与之前载入的一致，因此，当版本不一致时，将报告该错误。
 
 * Error Number: 8020
 
-    表被锁 
+    当表被加锁时，如果对该表执行写入操作，将出现该错误。 请将表解锁后，再进行尝试写入。
```
Comment (bb7133) [inline reference/error-codes.md:44]:
```suggestion
    当表被加锁时，如果对该表执行写入操作，将出现该错误。请将表解锁后，再进行尝试写入。
```

File: reference/error-codes.md [hunk 2]
```diff
@@ -33,83 +33,75 @@ TiDB 兼容 MySQL 的错误码，在大多数情况下，返回和 MySQL 一样
...
 
 * Error Number: 8025
 
-    写入的单条键值对过大 
+    写入的单条键值对过大。TiDB 最大支持 6MB 的单个键值对，超过该限制需要将过大的单行数据进行人工处理，以满足 6MB 限制。
 
 * Error Number: 8026
 
-    没有实现的接口 
+    使用了没有实现的接口函数。属于内部错误，不会返回给应用程序。
```
Comment (bb7133) [inline reference/error-codes.md:68]: Users may get confused on what '不会返回给应用程序' mean...Do they need to contact us, if such error code are returned to us?

**Comment by bb7133** [comment]
Rest LGTM, this update makes the error description much better.

File: reference/error-codes.md [hunk 3]
```diff
@@ -33,83 +33,75 @@ TiDB 兼容 MySQL 的错误码，在大多数情况下，返回和 MySQL 一样
...
```
Comment (jackysp) [inline reference/error-codes.md:68]:
```suggestion
    使用了没有实现的接口函数。该错误仅用于数据库内部，应用程序不会收到这个错误。
```

File: reference/error-codes.md (hunk 2 above)
Comment (jackysp) [reply reference/error-codes.md]: Changed a little, PTAL

**Comment by jackysp** [comment]
Updated, PTAL @TomShawn .

File: reference/error-codes.md [hunk 4]
```diff
@@ -33,83 +33,75 @@ TiDB 兼容 MySQL 的错误码，在大多数情况下，返回和 MySQL 一样
...
 
 * Error Number: 8030
 
-    转变为带符号正整数时发生了越界，显示为负数 
+    将值转变为带符号正整数时发生了越界，将结果显示为负数。多在告警信息里出现。
 
 * Error Number: 8031
 
-    负数转变为无符号数时，被转变为正数 
+    将负数转变为无符号数时，被负数转变为正数。多在告警信息里出现。
```
Comment (TomShawn) [inline reference/error-codes.md:88]:
```suggestion
    将负数转变为无符号数时，将负数转变为了正数。多在告警信息里出现。
```

File: reference/error-codes.md [hunk 5]
```diff
@@ -33,83 +33,75 @@ TiDB 兼容 MySQL 的错误码，在大多数情况下，返回和 MySQL 一样
...
 
 * Error Number: 8027
 
-    表结构版本过期 
+    表结构版本过期。TiDB 使用 F1 的在线 Schema 变更算法来执行 DDL。当 TiDB server 表结构版本落后于整个系统的时，执行 SQL 将遇到该错误。遇到该错误，请检查该 TiDB server 与 PD leader 之间的网络。
 
 * Error Number: 8028
 
-    表结构发生了变化 
+    TiDB 没有表锁（在 MySQL 中称为元数据锁，在某些其他数据库中称为意向锁等等）。当事务执行时，TiDB 表结构发生了变化是无法被事务感知到的。因此，TiDB 在事务提交时，会对事务涉及表的结构进行检查。如果事务执行中表结构发生了变化，则事务将提交失败，并返回该错误。遇到该错误，应用程序可以安全地重新执行整个事务。
```
Comment (TomShawn) [inline reference/error-codes.md:76]:
```suggestion
    TiDB 没有表锁（在 MySQL 中称为元数据锁，在其他数据库中可能称为意向锁）。当事务执行时，TiDB 表结构发生了变化是无法被事务感知到的。因此，TiDB 在事务提交时，会对事务涉及表的结构进行检查。如果事务执行中表结构发生了变化，则事务将提交失败，并返回该错误。遇到该错误，应用程序可以安全地重新执行整个事务。
```

**Pull request merged**